cmd/ambex/main.go

```diff
@@ -506,12 +613,13 @@ func update(ctx context.Context, config v2cache.SnapshotCache, configv3 v3cache.
 	// This used to just directly update envoy. Since we want ratelimiting, we now send an
 	// Update object down the channel with a function that knows how to do the update if/when
 	// the ratelimiting logic decides.
-	//
-	// We also need to pay attention to contexts here so we can shutdown properly. If we didn't
-	// have the context portion, the ratelimit goroutine could shutdown first and we could end
-	// up blocking here and never shutting down.
-	select {
-	case updates <- Update{version, func() error {
+
+	log.Debugf("Created snapshot %s", version)
+	csDump(snapdirPath, numsnaps, curgen, &snapshot, &snapshotv3)
```

Comment:
probably fine for hotfix but should probably not call this function for reals unless in some form of debug mode.

Reply:
As discussed: we want it to run all the time. If needed you can set `AMBASSADOR_AMBEX_SNAPSHOT_COUNT=0` to disable it. 🙂